tools/src/main/java/org/apache/kafka/tools/StreamsCleanupClient.java

```diff
@@ -0,0 +1,248 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership. The ASF licenses this file
+ * to You under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package org.apache.kafka.tools;
+
+import joptsimple.OptionException;
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+import kafka.admin.TopicCommand;
+import kafka.utils.ZkUtils;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.security.JaasUtils;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * {@link StreamsCleanupClient} resets a Kafka Streams application to the very beginning. Thus, an application can
```

Comment:
"... resets a Kafka Streams application to the very beginning." sounds a bit unclear IMHO.

Perhaps (just some ideas):

> {@link StreamsCleanupClient} resets/cleans the processing state of a Kafka Streams application so that, for example, you can reprocess its input from scratch.
>
> Resetting the processing state of an application includes the following actions:
> 1. setting the application's consumer offsets for source and internal topics to zero;
> 2. deleting any topics created internally by Kafka Streams for this application such as internal changelog topics for state stores.
>
> Topics explicitly created by the user -- e.g. source/input topics, intermediate topics created via `through()`, or output topics written to via `to()` -- will not be deleted or modified by this tool.